### PR TITLE
system76-power & system76-scheduler: pkgs.linuxPackages -> pkgs

### PR DESCRIPTION
**Pull request by sjmonson**
## Description of changes

Move `system76-power` and `system76-scheduler` out of the kernel modules section. Neither of these packages depend on out-of-tree kernel modules.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by cmm** [comment]
I'm slightly confused by the `linuxPackages` attrset: is it for linux-specific packages or (more narrowly) linux-kernel-specific packages?  If it's the latter, then this change makes perfect sense.

**Comment by sjmonson** [comment]
I believe it is the latter. When you put a package in ‘linuxPackages’ you get a version for every kernel. That makes sense if the package builds a dkms module or depends on specific kernels. Neither of these packages have that problem.

**Review by cmm — APPROVED** [review]

**Comment by sjmonson** [comment]
Sorry to bump this but who is responsible for merging? The maintainer has already given approval. I would like to start a new PR to get `system76-power` updated as its very out of date, but this needs to be in first.

**Comment by ahoneybun** [comment]
@sjmonson once that is done is there anything I can do to help with getting system76-dkms updated? I'm working on getting a new tagged release out.

**Comment by alyssais** [comment]
Usually for package moves like this we'd add an alias.  There should be some examples already in linux-kernels.nix.

**Comment by ahoneybun** [comment]
There is a new tag for system76-dkms:

https://github.com/pop-os/system76-dkms/releases/tag/1.0.16

**Comment by alyssais** [comment]
Still needs an alias.

**Comment by ahoneybun** [comment]
Not sure what that is in this context but perhaps @sjmonson knows.

File: pkgs/top-level/linux-kernels.nix
Comment (lilyinstarlight) [inline pkgs/top-level/linux-kernels.nix:535]: Move these down to the section at line 604 that is conditional on `config.allowAliases` 
 to fix eval cleanliness and use `pkgs.system76-power` instead of `(callPackage ...)`

Comment (sjmonson) [reply pkgs/top-level/linux-kernels.nix]: Ack

**Review by cmm — APPROVED** [review]

**Comment by alyssais** [comment]
system76-power is failing to build, but that predates this change, so let's go ahead.

**Comment by alyssais** [comment]
Would you mind squashing the aliases for each package into the commits that move the attributes?  It's nice if there isn't an in-between state where the alias isn't there.

**Comment by sjmonson** [comment]
> system76-power is failing to build, but that predates this change, so let's go ahead.

Yeah that was my bad. I pulled the alias changes out of my stash and forgot they predate moving the packages to `pkgs/by-name`.

> Would you mind squashing the aliases for each package into the commits that move the attributes? It's nice if there isn't an in between state where the alias isn't there.

Done

**Review by alyssais — APPROVED** [review]

**Comment by ahoneybun** [comment]
> system76-power is failing to build, but that predates this change, so let's go ahead.

Is that connected to this?

```
building the system configuration...
error: builder for '/nix/store/6ang6g3hkygpymgz6kggl3bvamxvyh17-system76-power-1.1.23.drv' failed with exit code 101;
       last 10 log lines:
       > error: failed to get `sysfs-class` as a dependency of package `system76-power v1.1.23 (/build/source)`
       >
       > Caused by:
       >   failed to load source for dependency `sysfs-class`
       >
       > Caused by:
       >   Unable to update https://github.com/pop-os/sysfs-class#ab63e7f6
       >
       > Caused by:
       >   can't checkout from 'https://github.com/pop-os/sysfs-class': you are in the offline mode (--offline)
       For full logs, run 'nix log /nix/store/6ang6g3hkygpymgz6kggl3bvamxvyh17-system76-power-1.1.23.drv'.
error: 1 dependencies of derivation '/nix/store/3awmhvd9i523ayhclqmcs8dqkjx09bps-dbus-1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/90scmy4mz557sa6f9j1zgyfbc29lylib-system-path.drv' failed to build
error (ignored): error: cannot unlink '/tmp/nix-build-nvidia-x11-560.35.03-6.10.14.drv-1/build/NVIDIA-Linux-x86_64-560.35.03': Directory not empty
error: 1 dependencies of derivation '/nix/store/4dwfv5rmwcjl4mly2ca5w47qw822ydbd-unit-system76-power.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/1rgbfk416ayb4jn3fa27amc0jl0bx360-nixos-system-garrus-24.11.20241014.a3c0b3b.drv' failed to build
``` 

or something else?

**Comment by sjmonson** [comment]
@ahoneybun No something else. This PR does not update system76-power. I have a follow-on PR planned that will update the package. However, I have never seen that issue.

**Pull request merged**
